pkg/cephmgr/osd/agent.go

```diff
@@ -421,16 +623,21 @@ func (a *osdAgent) createOrStartOSD(context *clusterd.Context, connection client
 			return fmt.Errorf("failed to initialize OSD at %s: %+v", config.rootPath, err)
 		}
 
-		// save the osd to applied state
+		// save the osd information to applied state
+		dataDiskUUID, metadataDiskUUID := "", ""
+		if config.partitionScheme != nil {
```

Comment:
in the case of a filestore OSD being created for a desired directory, there would be no partition scheme here, so the nil check is necessary.